### ISSUE TITLE
VPN on Android does not connect, stuck on "attempting to confirm connection"

### ISSUE THREAD

**Issue by JKingZapata** [mozilla-mobile/mozilla-vpn-client#1495]
Users on Android devices are unable to connect to Mozilla VPN version 2.4. VPN shows "attempting to confirm connection" and after a moment the switch turns off.

**Technical Information**

* OS version: Android 8.1, Android 9
* Device model: Alcatel 3T8, Moto G6, Samsung J3
* FX product Mozilla VPN 2.4

**Steps to Reproduce**

1. Sign up to a Mozilla VPN subscription
1. Download/install Mozilla VPN
1. Flip switch to connect to VPN

**Actual Results**

1. Sign up to a Mozilla VPN subscription
1. Download/install Mozilla VPN
1. Flip switch to connect to VPN
1. Switch shows "attempting to confirm connection"
1. Switch never connects and turns off after a moment

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-845)

**Comment by strseb** [comment]
Probably a duplicate of  #637 :/

**Comment by JKingZapata** [comment]
Attaching logs

[VPN 2.4 connection issue Android.txt](https://github.com/mozilla-mobile/mozilla-vpn-client/files/6932670/VPN.2.4.connection.issue.Android.txt)

**Comment by birdsarah** [comment]
Also #1265

**Comment by LuminitaB** [comment]
We have checked this today on the following setup:
- Samsung Galaxy S10 Plus, with Android 9
- Mozilla VPN app v. 2.5.0 (16293690)
The VPN does not connect, it stays in the "Attempting to confirm connection" status and it throws the "No internet connection try Again" message. (See below image)

![Screenshot_20210820-104127](https://user-images.githubusercontent.com/52028275/130201655-82a573b7-2a44-4010-9dc1-bd538a9106c8.jpg)
[Mozilla VPN logs.docx](https://github.com/mozilla-mobile/mozilla-vpn-client/files/7020076/Mozilla.VPN.logs.docx)

I have also attached the logs from the app:
[Mozilla VPN logs.docx](https://github.com/mozilla-mobile/mozilla-vpn-client/files/7020081/Mozilla.VPN.logs.docx)

**Comment by ValentinaPC** [comment]
I encountered the "Connecting..." issue on both Windows and Linux, and forced me to restart the PC so that I can use it again.
Here are logs from Windows.
[mozillavpn-2021-8-20.txt](https://github.com/mozilla-mobile/mozilla-vpn-client/files/7020301/mozillavpn-2021-8-20.txt)

**Comment by oskirby** [comment]
> I encountered the "Connecting..." issue on both Windows and Linux, and forced me to restart the PC so that I can use it again.
> Here are logs from Windows.
> [mozillavpn-2021-8-20.txt](https://github.com/mozilla-mobile/mozilla-vpn-client/files/7020301/mozillavpn-2021-8-20.txt)

@ValentinaPC please file that as a separate issue, since it seems like a something different from what's been reported here.

**Comment by ValentinaPC** [comment]
@oskirby seems like I cannot reproduce the issue anymore. If I will, I'll log it separately. Thanks!

**Comment by JKingZapata** [comment]
60 tickets with this issue as of 08/24

**Comment by strseb** [comment]
Closing this as valentina can't reproduce :)

**Comment by ValentinaPC** [comment]
@strseb I was referring to the separate issue I found on Windows and Linux, that @oskirby asked me to log it separately.
The initial thing is still happening, as confirmed by @LuminitaB.

**Comment by LuminitaB** [comment]
I have noticed that, killing the app while in the "Attempting to confirm connection" state and reopening it, it show the VPN as ON, but no traffic seems to go through. (please see below video)
![2021-09-03_17h53_55](https://user-images.githubusercontent.com/52028275/132025876-3f5997c8-9a16-4948-9128-c8a60b1c5312.gif)

**Comment by data-sync-user** [comment]
➤ Rebecca Billings commented:

The original bug is fixed, Luminita Buruian  please file your other issue as a separate bug.

**Comment by data-sync-user** [comment]
➤ Luminita Buruian commented:

Rebecca Billings Thank you for the above info.

I have just verified the issue with the latest Android build from main and latest Android builds for 2.5 release and I cannot connect to the VPN, neither on android 8, nor on android 9. The same issue occurs, the status gets stuck on "Connecting. Attempting to confirm connection" and after a while the VPN turns off.

The scenario in my latest comment is not a separate issue, it is related just another discovery related to the current issue.

CC: Valentina Virlics

**Comment by data-sync-user** [comment]
➤ Basti (Sebastian) Streich commented:

Hi! Luminita Buruian - could you send logs again? 🙂 On the new build we have added a bit more debug info to see what is going wrong

**Comment by data-sync-user** [comment]
➤ Luminita Buruian commented:

Hello Basti (Sebastian) Streich , sure thing.

Please see attached logs from Mozilla VPN 2.5.0 (16311240) using an Android 9 Samsung Galaxy S10 Plus.

Please let me know if I can be of any more assistance.

[^Mozilla VPN loggs - stuck on attempting to confirm connection.txt]

**Comment by data-sync-user** [comment]
➤ Basti (Sebastian) Streich commented:

[^qa_more_ping_logs.apk]

Luminita Buruian  - Thank you for the logs, can you retry with this apk and get logs? This should produce even more logs, that would tell me why our fallbacks are not catching this case. 🙂

**Comment by data-sync-user** [comment]
➤ Luminita Buruian commented:

Basti (Sebastian) Streich Happy to help!

Please see below the requested logs. This time the app itself got stuck/frozen and nothing worked, until the OS reported that the app is not responding and only after that I was able to reopen the app to get the logs.

[^Mozilla VPN logs - stuck on attempting to confirm connection(2).txt]

I hope this helps.

Thank you.

**Comment by data-sync-user** [comment]
➤ Basti (Sebastian) Streich commented:

Hi! Luminita Buruian  - Sorry the .apk was not meant to get stuck, hope i figured that out. Can you give it a try?

[^qa_ping_v2.apk]

**Comment by data-sync-user** [comment]
➤ Luminita Buruian commented:

Hello Basti,

I used the latest build you provided and, unfortunately, the behavior seems to be the same: VPN status stuck on "Attempting to confirm connection" state, the app freezes until the OS reports that the app is not responding. See screenshot below.

!Screenshot_20210921-120022.jpg|width=1440,height=3040!

I have also got the logs.

[^Mozilla VPN logs - stuck on attempting to confirm connection - 09.21.txt]

I hope this helps.

Thank you.

**Comment by data-sync-user** [comment]
➤ Basti (Sebastian) Streich commented:

Hello! Thank you the logs were super useful! - It just ruled out a big chunk of possible errors. Can you try this one? As always thank you so much for your help!

[^qa_ping_exit_code.apk]

**Comment by data-sync-user** [comment]
➤ Luminita Buruian commented:

Hello, happy to help.

The newest build:

  * I was able to turn on VPN and it connected pretty quick

!Screenshot_20210921-165528.jpg|width=1440,height=3040!

  * I saw no traffic in the status screen though

!Screenshot_20210921-165522.jpg|width=1440,height=3040!

  * The app froze while tryin to get the logs, like the other before, until the OS reported that the app is not responding. Killing the app did not help.

Here are the logs: 

[^Mozilla VPN logs - stuck on attempting to confirm connection - 09.21(2).txt]

Let me know if there is anything else.

Thank you.

**Comment by data-sync-user** [comment]
➤ Luminita Buruian commented:

Verified as fixed on Mozilla VPN 2.6.0 (16331130), while using using an Android 9 Samsung Galaxy S10 Plus.

The VPN successfully connects, but I do not see any traffic in the CI screen. 

Basti (Sebastian) Streich let me know if there is anything else I need to check.

Thank you.

**Comment by JC-Support** [comment]
Verified as fixed on Mozilla VPN 2.6.0 (16331130), while using using an Android 9 Samsung Galaxy S10 Plus.
